File: postgresql/net9/Controllers/Test1Controller.cs
using Microsoft.AspNetCore.Mvc;
using net9.Models.Requests;
using net9.Services;

namespace net9.Controllers {

    [ApiController]
    [Route("api/v1/test1")]
    public class Test1Controller: ControllerBase
    {
        private readonly ITest1Service _test1Service;
        public Test1Controller(ITest1Service test1Service) {
            _test1Service = test1Service;
        }

        [HttpGet("{id}")]
        public async Task<IActionResult> GetById([FromRoute] int id){
            var response = await _test1Service.GetById(id);
            return StatusCode(response.HttpStatusCode, response.BodyResponse);
        }

        [HttpPost]
        public async Task<IActionResult> Create([FromBody] CreateRequest createRequest){
            var response = await _test1Service.Create(createRequest);
            return StatusCode(response.HttpStatusCode, response.BodyResponse);
        }

        [HttpPut]
        public async Task<IActionResult> Update([FromBody] UpdateRequest updateRequest)
        {
            var response = await _test1Service.Update(updateRequest);
            return StatusCode(response.HttpStatusCode, response.BodyResponse);
        }

        [HttpDelete]
        public async Task<IActionResult> Delete([FromBody] DeleteRequest deleteRequest)
        {
            var response = await _test1Service.Delete(deleteRequest);
            return StatusCode(response.HttpStatusCode, response.BodyResponse);
        }
    }
}

File: redis/rust/src/controllers/test1_controller.rs
use std::sync::Arc;
use axum::extract::{Path, State};
use axum::Json;
use axum::response::IntoResponse;
use crate::models::requests::create_request::CreateRequest;
use crate::models::requests::delete_request::DeleteRequest;
use crate::services::test1_service::Test1Service;
use crate::states::app_state::AppState;

pub async fn create_handler(State(state): State<Arc<AppState>>, Json(create_request): Json<CreateRequest>) -> impl IntoResponse {
    let response = state.test1_service.lock().await.create(create_request).await;
    (response.http_status_code, Json(response.body_response))
}

pub async fn get_handler(State(state): State<Arc<AppState>>, Path(id): Path<String>) -> impl IntoResponse {
    let response = state.test1_service.lock().await.get(&id).await;
    (response.http_status_code, Json(response.body_response))
}

pub async fn delete_handler(State(state): State<Arc<AppState>>, Json(delete_request): Json<DeleteRequest>) -> impl IntoResponse {
    let response = state.test1_service.lock().await.delete(delete_request).await;
    (response.http_status_code, Json(response.body_response))
}